mocks/firestore.js

```diff
@@ -249,6 +249,10 @@ FakeFirestore.DocumentReference = class {
   startAt() {
     return this.query.startAt(...arguments);
   }
+
+  withConverter() {
+    return this.query.withConverter(...arguments);
+  }
```

Comment:
It looks like you can call `withConverter` on the collection as well as the document.  Looking at this example, `withCoverter` is called before the `doc` is fetched:
https://firebase.google.com/docs/reference/js/firebase.firestore.FirestoreDataConverter

Could we add this to `CollectionReference` as well?  Then add a test that asserts we can call `withConverter` before `doc` _and_ after?

Reply:
Thanks for clarifying.
Could we still get one test with an example like the one I linked in the documentation?

Reply:
Added `DocumentReference `implementation with tests to cover some cases. The tests being in so many different places for firestore functionality is a bit odd as to where to add them so if needed they can be moved

Reply:
Thank you! And that's a very fair point. I think cleaning up our tests so it's easier for new contributors would be good